frame/im-online/src/lib.rs

```diff
@@ -485,11 +446,10 @@ pub mod pallet {
 			if let (false, Some(public)) = (exists, public) {
 				Self::deposit_event(Event::<T>::HeartbeatReceived { authority_id: public.clone() });
 
-				let network_state_bounded = BoundedOpaqueNetworkState::<
-					T::MaxPeerDataEncodingSize,
-					T::MaxPeerDataEncodingSize,
-					T::MaxPeerInHeartbeats,
-				>::force_from(&heartbeat.network_state);
+				let network_state_bounded =
+					BoundedOpaqueNetworkState::<T::MaxPeerDataEncodingSize>::force_from(
+						&heartbeat.network_state,
+					);
```

Comment:
We should get rid of the entire network_state. In the heartbeat and everywhere in this pallet.